src/NexusMods.DataModel/Diagnostics/DiagnosticManager.cs

```diff
@@ -1,190 +1,63 @@
-    /// Constructor.
-    /// </summary>
     public DiagnosticManager(
         ILogger<DiagnosticManager> logger,
-        IDataStore dataStore,
-        IOptionsMonitor<DiagnosticOptions> optionsMonitor,
-        IEnumerable<ILoadoutDiagnosticEmitter> loadoutDiagnosticEmitters,
-        IEnumerable<IModDiagnosticEmitter> modDiagnosticEmitters,
-        IEnumerable<IModFileDiagnosticEmitter> modFileDiagnosticEmitters)
+        ILoadoutRegistry loadoutRegistry)
     {
         _logger = logger;
-        _dataStore = dataStore;
-        _optionsMonitor = optionsMonitor;
-
-        _loadoutDiagnosticEmitters = loadoutDiagnosticEmitters.ToArray();
-        _modDiagnosticEmitters = modDiagnosticEmitters.ToArray();
-        _modFileDiagnosticEmitters = modFileDiagnosticEmitters.ToArray();
-
-        _compositeDisposable = new CompositeDisposable();
-
-        // Listen to option changes and update the currently existing diagnostics.
-        var disposable = _optionsMonitor.OnChange(newOptions =>
-        {
-            var filteredDiagnostics = FilterDiagnostics(_diagnosticCache.Items, newOptions);
-            _diagnosticCache.Edit(updater =>
-            {
-                updater.Clear();
-                updater.AddRange(filteredDiagnostics);
-            });
-        });
-
-        if (disposable is not null) _compositeDisposable.Add(disposable);
-    }
-
-    public IObservable<IChangeSet<Diagnostic>> DiagnosticChanges => _diagnosticCache.Connect();
-    public IEnumerable<Diagnostic> ActiveDiagnostics => _diagnosticCache.Items;
-
-    public async ValueTask OnLoadoutChanged(Loadout loadout)
-    {
-        await RefreshLoadoutDiagnostics(loadout);
-        RefreshModDiagnostics(loadout);
-    }
-
-    public void ClearDiagnostics()
-    {
-        _diagnosticCache.Edit(updater => updater.Clear());
+        _loadoutRegistry = loadoutRegistry;
     }
 
-    internal async Task RefreshLoadoutDiagnostics(Loadout loadout)
+    public IObservable<Diagnostic[]> GetLoadoutDiagnostics(LoadoutId loadoutId)
     {
-        // Remove outdated diagnostics for previous revisions of the loadout
-        RemoveDiagnostics(kv => kv.DataReferences.Values
-            .OfType<LoadoutReference>()
-            .Any(loadoutReference =>
-                loadoutReference.DataId == loadout.LoadoutId
-                && !Equals(loadoutReference.DataStoreId, loadout.DataStoreId)
-            )
-        );
-
-        var newDiagnostics = await _loadoutDiagnosticEmitters
-            .SelectAsync(async emitter => await emitter.Diagnose(loadout).ToArrayAsync())
-            .ToArrayAsync();
+        // TODO: cancellation token
+        var cancellationToken = CancellationToken.None;
 
-        AddDiagnostics(newDiagnostics.SelectMany(vals => vals));
+        return _loadoutRegistry
+            .RevisionsAsLoadouts(loadoutId)
+            .DistinctUntilChanged(loadout => loadout.DataStoreId)
+            .Throttle(dueTime: TimeSpan.FromMilliseconds(250))
```

Comment:
Use `Debounce` rather than `Throttle` https://reactivex.io/documentation/operators/debounce.html

I think it's more suited to our needs, as it always emits the last value after a while, rather than potentially ignoring it as happens with `Throttle`